### ISSUE TITLE
Volumes being created without my consent

### ISSUE THREAD

**Issue by kingbuzzman** [docker/compose#5210]
With normal docker:

```
$ docker volume ls
DRIVER              VOLUME NAME
```

Run the server
`docker run -it --rm -p 6379:6379 redis:3.2-alpine`

Save some stuff
`printf "set key value\nsave" | redis-cli`

Exactly what i expect.
```
$ docker volume ls
DRIVER              VOLUME NAME
```

----------

Run the server
`docker run -it --rm -p 6379:6379 -v /data redis:3.2-alpine`

Save some stuff
`printf "set key value\nsave" | redis-cli`

Cool, the data was saved to a random volume -- cool.
```
docker volume ls
DRIVER              VOLUME NAME
local               47dddf3c80d654478ba8577f02d526668f09a9937c615b48515f7d2870731d08
```

----------

Why is it that this:
```
services:
  redis-ns:
    restart: always
    image: redis:3.2-alpine
```

is equal to this: `docker run -it --rm -p 6379:6379 -v /data redis:3.2-alpine` when it should be more equal to this: `docker run -it --rm -p 6379:6379 redis:3.2-alpine`

The frustrating part is that i have 20+ unnamed volumes (every time i restart redis i get a new one (and i run 2 instances!)) that i rather not see there.

**TLDR** docker-compose adding volumes even though you never wanted them

**Comment by shin-** [comment]
It's an [advertised feature](https://docs.docker.com/compose/overview/#preserve-volume-data-when-containers-are-created) of `docker-compose`.

**Comment by kingbuzzman** [comment]
@shin- thats great, how do i turn it off? It's frustrating me beyond belief

**Comment by shin-** [comment]
There's currently no way to opt out of the feature directly. If you don't actually declare anonymous volumes in your Compose file, you might want to make sure the images you use don't do so in their Dockerfile. Volumes are only ever created if they are declared in the first place.

**Comment by kingbuzzman** [comment]
@shin- Darn, i really didnt want to maintain my own images of redis (im using the official version) -- ok, i guess thats a workaround. I'll see how much time i have and i'll make a little PR to undo it. Thanks!

**Comment by abraj** [comment]
You can instead use [redis/redis-stack-server](https://hub.docker.com/r/redis/redis-stack-server) (by redis) instead.
Unlike official redis image, It does not automatically create anonymous volumes.
